JoonasHeinonen/escape-from-zordoom-prison
# Feature/joonas expand test level

## [Expand test level](https://trello.com/c/nFI4M3TR/91-expand-text-level)

**Girdeux Dialogic - Passive**
 - https://github.com/JoonasHeinonen/escape-from-zordoom-prison/assets/33086486/856ca1b5-3929-40b7-bca8-9fb75e00b0a0

**Girdeux AI - Wall jump**
 - https://github.com/JoonasHeinonen/escape-from-zordoom-prison/assets/33086486/1bfafe40-2d8c-4106-9c66-43a6fd3cfb26

**Mia**
 - https://github.com/JoonasHeinonen/escape-from-zordoom-prison/assets/33086486/b00355fc-cfee-45f7-8b84-2185b3de35eb

**Static enemy and ammo (_Ammo WIP_)**
 - https://github.com/JoonasHeinonen/escape-from-zordoom-prison/assets/33086486/5890fd8d-1b2f-49f9-ba22-77641aaf6b2c

**Door fix**
 - https://github.com/JoonasHeinonen/escape-from-zordoom-prison/assets/33086486/9067e23d-a342-4011-b6b1-a36abb642fbe

## Changes:
 - Created the teleportation system and implemented the logic on defining the point 1 and point 2.
 - Imported a placeholder model for the teleports.
 - Intended some poorly formatted code.
 - Updated the PR template for better pull requests.
 - Created some new blocks for the game.
 - Expanded the elements enum for the enemies.
 - Expanded the dialogic options and optimized the code used for it.
 - Created a new NPC, Mia.
 - Resized the old talking heads from their original sizes to a fixed 200 x 200 pixels.
 - Boss logic improved: No teleports nor doors won't open until the boss is defeated.
 - Ammo and health logic finally implemented.

## How to test:
 - See that the teleportation and doors work. But see that when you have an active boss fight, they won't.
 - See that all the NPC's work as intended. (All the others should be the same, Grokes will turn at you once shot at.)
 - See that the dialogic looks good and that the talking heads are good.
 - See that the dialogic by Girdeux is good enough.
 - See that you will die after losing enough health.
 - See that you can still restore health from the Nanotech nodes.
 - See that you will run out of ammo for any or all of your weapons.
 - See that you can get back your ammunition.

## Sources cited:
 - [Remove last character from variable (a string)](https://www.reddit.com/r/godot/comments/h8a2nc/remove_last_character_from_variable_a_string/)
